x-pack/platform/plugins/shared/onechat/public/plugin.tsx

```diff
@@ -84,21 +84,28 @@ export class OnechatPlugin
 
   start(core: CoreStart, startDependencies: OnechatStartDependencies): OnechatPluginStart {
     const { http } = core;
+    const { licensing, inference } = startDependencies;
     docLinks.setDocLinks(core.docLinks.links);
 
     const agentService = new AgentService({ http });
     const chatService = new ChatService({ http });
     const conversationsService = new ConversationsService({ http });
     const toolsService = new ToolsService({ http });
+    const accessChecker = new AgentBuilderAccessChecker({ licensing, inference });
 
-    this.internalServices = {
-      agentService,
-      chatService,
-      conversationsService,
-      toolsService,
-      startDependencies,
+    const getInternalServices = async () => {
+      return {
+        agentService,
+        chatService,
+        conversationsService,
+        toolsService,
+        startDependencies,
+        access: await accessChecker.checkAccess(),
```

Comment:
I'm not sure if the function `checkAccess` can possibly throw errors or not, but I would wrap this in a `try/catch` just in case to not have a chance of an unhandled promise rejection.